ios/browser/api/ads/brave_ads.mm

```diff
@@ -266,6 +244,12 @@ - (void)initialize:(void (^)(bool))completion {
   adsClient = new AdsClientIOS(self);
   adsClientNotifier = new brave_ads::AdsClientNotifier();
   ads = brave_ads::Ads::CreateInstance(adsClient);
+  ads->SetSysInfo(brave_ads::mojom::SysInfo::New(
+      base::SysNSStringToUTF8(sysInfo.deviceId)));
```

Comment:
Did `sysInfo.cppObjPtr` not work instead of creating a new one?

Reply:
Fixed

Reply:
added `#import "brave/ios/browser/api/ads/ads.mojom.objc+private.h"`